sim-lib/src/lib.rs

```diff
@@ -71,3 +114,30 @@ struct PaymentResult {
     settled: bool,
     action: u8,
 }
+
+pub struct Simulation {
+    // The lightning node that is being simulated.
+    nodes: HashMap<String, Arc<dyn LightningNode>>,
+
+    // The activity that are to be executed on the node.
+    activity: Vec<ActivityDefinition>,
+}
+
+impl Simulation {
+    pub fn new(
+        nodes: HashMap<String, Arc<dyn LightningNode>>,
```

Comment:
`String` -> `Pubkey` per @sr-gi's suggestion

Reply:
Yep, this is missing